File: Stage23/Java/Problem1931.java
package Stage23.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Problem1931 {
    static class Meeting {
        int start, end;
        Meeting(int start, int end) {
            this.start = start;
            this.end = end;
        }
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        Meeting[] arr = new Meeting[N];

        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int s = Integer.parseInt(st.nextToken());
            int e = Integer.parseInt(st.nextToken());
            arr[i] = new Meeting(s, e);
        }

        Arrays.sort(arr, (a, b) -> {
            if (a.end == b.end) return a.start - b.start;
            return a.end - b.end;
        });

        int count = 0, endTime = 0;

        for (Meeting m : arr) {
            if (m.start >= endTime) {
                count++;
                endTime = m.end;
            }
        }

        System.out.println(count);
        br.close();
    }
}

File: Stage23/C/Problem1931.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int start;
    int end;
} Meeting;

int compare(const void *a, const void *b) {
    Meeting *m1 = (Meeting *)a;
    Meeting *m2 = (Meeting *)b;

    if (m1 -> end == m2 -> end)
        return m1 -> start - m2 -> start;
    return m1 -> end - m2 -> end;
}

int main(void) {
    int N;
    scanf("%d", &N);

    Meeting arr[N];
    for (int i = 0; i < N; i++) {
        scanf("%d %d", &arr[i].start, &arr[i].end);
    }

    qsort(arr, N, sizeof(Meeting), compare);

    int count = 0;
    int endTime = 0;

    for (int i = 0; i < N; i++) {
        if (arr[i].start >= endTime) {
            count++;
            endTime = arr[i].end;
        }
    }

    printf("%d\n", count);
    return 0;
}
